charts/kube-prometheus-stack/templates/prometheus/clusterrole.yaml

```diff
@@ -24,4 +24,7 @@ rules:
   verbs: ["get", "list", "watch"]
 - nonResourceURLs: ["/metrics", "/metrics/cadvisor"]
   verbs: ["get"]
+{{- if and .Values.prometheus.enabled .Values.prometheus.additionalRulesForClusterRole }}
```

Comment:
`.Values.prometheus.enabled` is checked at the top of the file, so it isn't needed here.

Reply:
Unnecessary check has been removed.